### ISSUE TITLE
Copilot integration

### ISSUE THREAD

**Issue by gruppo4** [dbeaver/dbeaver#20072]
I wondered if a copilot integration is on its way.

Me and my company opted for using copilot instead of chatGpt. Currently is the faster and reliable service we tried and the subscription plan is fair enough. We work a lot better.

Needless to say, we don't use dbeaver anymore to write queries, but we create the query with vscode or IntelliJ Idea and then copy/paste onto dbeaver.

If we could use copilot directly inside the eclipse editor of dbeaver, it would be fantastic.

Note that JetBrains has his own integration for databases, but I don't like it, I love dbeaver.

**Comment by E1izabeth** [comment]
Thank you for the feature request.
We have it in our plans, but I can't say, when exactly it will be implemented.

**Comment by VasantSachdewa** [comment]
@E1izabeth is there any way I can participate and contribute

**Comment by E1izabeth** [comment]
We're always open for contributions. You may find more info about how to contribute in DBeaver [here](https://github.com/dbeaver/dbeaver/blob/devel/.github/CONTRIBUTING.md)

**Comment by allanlaal** [comment]
@E1izabeth look into refact.ai - its an opensource clone of GH CoPilot

**Comment by luga97** [comment]
Same here, I guess this will help me a lot, we need this!

**Comment by jjkchung87** [comment]
+1 would love the functionality

**Comment by KodornaRocks** [comment]
This is the ONLY thing keeping me with the slow and overpriced datagrip

**Comment by deepaksenthilkumar** [comment]
+1

**Comment by deepaksenthilkumar** [comment]
Are you planning to include it anytime soon team?

**Comment by E1izabeth** [comment]
Currently, it is not planned for near future releases.

**Comment by alifemove** [comment]
Currently, it is the only reason I use Jetbrains Datagrip instead of DBeaver.

**Comment by DaggerJackfast** [comment]
+1

**Comment by mkhayom** [comment]
> Currently, it is the only reason I use Jetbrains Datagrip instead of DBeaver.
same with our


> This is the ONLY thing keeping me with the slow and overpriced datagrip

+1

**Comment by marcoseloso77** [comment]
The same here

> > Currently, it is the only reason I use Jetbrains Datagrip instead of DBeaver.
> > same with our
> 
> > This is the ONLY thing keeping me with the slow and overpriced datagrip
> 
> +1

+1

**Comment by alifemove** [comment]
This is currently the second most upvoted feature request. Surely it is worth prioritizing.
Correction: Is now the most upvoted feature request.

**Comment by guidogabriel** [comment]
+1

**Comment by mesutpiskin** [comment]
+1
up up

**Comment by bsillito** [comment]
+1

**Comment by mattheusrocha2** [comment]
+1

**Comment by Joao-S-Martins** [comment]
+1

**Comment by chimello** [comment]
+1

**Comment by FaigyL** [comment]
+1

**Comment by VVVHawk** [comment]
I join this request.

**Comment by susminv** [comment]
+1

**Comment by pfrozi** [comment]
+1

**Comment by mesutpiskin** [comment]
It seems that the suggestions here are not being considered. As a company, we have started using DataGrip, and I can say that its AI features work very well, improving our efficiency.

**Comment by alifemove** [comment]
Yeah, sad that its been the most requested feature for 2 months and the debeaver team is nowhere to be seen.

**Comment by slokkenquick** [comment]
1+

**Comment by ilkerhalil** [comment]
+1

**Comment by GuiSmith** [comment]
It would make things fast

**Comment by sidhant-khamankar** [comment]
If we just copy and paste the "update site URL" from [here](https://marketplace.eclipse.org/content/copilot4eclipse#maint-content-area) into Dbeaver's "Install new software => Work with field" then it installs the Copilot Chatbot. But again [this gets removed on updating Dbeaver](https://github.com/dbeaver/dbeaver/issues/5317)

**Comment by sidhant-khamankar** [comment]
https://marketplace.eclipse.org/content/github-copilot

Got error while trying to install official github copilot extension
```
Cannot complete the install because one or more required items could not be found.
  Software being installed: GitHub Copilot 0.2.0.202502180149 (com.microsoft.copilot.eclipse.feature.feature.group 0.2.0.202502180149)
  Missing requirement: com.microsoft.copilot.eclipse.core 0.2.0.202502180149 (com.microsoft.copilot.eclipse.core 0.2.0.202502180149) requires 'osgi.bundle; org.eclipse.lsp4e 0.18.12' but it could not be found
  Cannot satisfy dependency:
    From: GitHub Copilot 0.2.0.202502180149 (com.microsoft.copilot.eclipse.feature.feature.group 0.2.0.202502180149)
    To: org.eclipse.equinox.p2.iu; com.microsoft.copilot.eclipse.core [0.2.0.202502180149,0.2.0.202502180149]
```

**Comment by Destrolaric** [comment]
Completed by https://github.com/dbeaver/dbeaver/pull/37270

**Comment by alifemove** [comment]
@Destrolaric so it's available? How do we use it?

**Comment by Destrolaric** [comment]
Will be available as an installable plugin in the next release.
